### PR TITLE
Added Flight movement provider

### PR DESCRIPTION
**Pull request by Malcolmnixon**
This pull request fixes issues #79 and #80.

### CODE REVIEW

File: addons/godot-xr-tools/functions/Function_Flight_movement.gd [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+##     forwards direction as if the player had guide-fins or wings.
+##
+##     The Exclusive property specifies whether flight is exclusive (no further
+##     physics effects after flying) or whether additional effects such as 
+##     the default player gravity are applied.
+##
+
+
+# enum our buttons, should find a way to put this more central
+enum Buttons {
```
Comment (BastiaanOlij) [inline addons/godot-xr-tools/functions/Function_Flight_movement.gd:36]: @Malcolmnixon probably not for this PR but maybe a separate one, we should look into moving this into our util script so these enums are just available

**Comment by BastiaanOlij** [comment]
Looks great! Let me know if you're ready for this to be merged

**Comment by Malcolmnixon** [comment]
I think this one's ready to go in. Flight has such a diverse set of possibilities that I'm certain we'll get many requests for new features as people get a chance to play. Some we may be able to fit in, but many where it's better served by a game-specific customized script.

**Comment by BastiaanOlij** [comment]
thanks!

**Pull request merged**